### PR TITLE
Add back removed support for custom node builders in integration tests

### PR DESCRIPTION
**Pull request by zeptin**
This is needed for testing node features that are outside the SBFN codebase.

### CODE REVIEW

**Comment by zeptin** [comment]
@bokobza fair enough, I've pulled the changes out into some new methods and runners.

**Comment by fassadlr** [comment]
@zeptin this was removed because these custom nodes weren't being used anymore. If this is not something we are currently utilizing, why merge in unused code?

**Comment by bokobza** [comment]
It was in their code up to now, but since we now have a test lib anyone can use, why not do it there?

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs [hunk 1]
```diff
@@ -180,6 +181,16 @@ public CoreNode CloneStratisNode(CoreNode cloneNode)
             return node;
         }
 
+        public CoreNode CreateCustomPowNode(bool start, Action<IFullNodeBuilder> callback, Network network)
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs]: These 2 methods are similar aren't they?

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeRunners.cs [hunk 2]
```diff
@@ -217,4 +218,60 @@ public override void OnStart()
...
+        public CustomPowRunner(string dataDir, Action<IFullNodeBuilder> callback, Network network)
+            : base(dataDir)
+        {
+            this.callback = callback;
+            this.network = network;
+        }
+
+        public override void BuildNode()
+        {
+            var settings = new NodeSettings(this.network, args: new string[] { "-conf=bitcoin.conf", "-datadir=" + this.DataFolder }, loadConfiguration: false);
```
[the remark this answers is not in the document]
Comment (dangershony) [reply src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeRunners.cs]: Don't we have the config file name in the network?

**Comment by bokobza** [comment]
Looks good @zeptin 😃

**Pull request merged**